src/mrinufft/trajectories/display.py

```diff
@@ -19,186 +26,588 @@
     ax.axes.set_xlim3d(left=-KMAX, right=KMAX)
     ax.axes.set_ylim3d(bottom=-KMAX, top=KMAX)
     ax.axes.set_zlim3d(bottom=-KMAX, top=KMAX)
     ax.set_box_aspect((2 * KMAX, 2 * KMAX, 2 * KMAX))
-    ax.set_xlabel("kx", fontsize=FONTSIZE)
-    ax.set_ylabel("ky", fontsize=FONTSIZE)
-    ax.set_zlabel("kz", fontsize=FONTSIZE)
+    ax.set_xlabel("kx", fontsize=displayConfig.fontsize)
+    ax.set_ylabel("ky", fontsize=displayConfig.fontsize)
+    ax.set_zlabel("kz", fontsize=displayConfig.fontsize)
     return ax
 
 
+######################
+# TRAJECTORY DISPLAY #
+######################
+
+
 def display_2D_trajectory(
     trajectory,
-    size=5,
+    figsize=5,
     one_shot=False,
-    constraints=False,
     subfigure=None,
-    resolution=DEFAULT_RESOLUTION,
+    show_constraints=False,
+    gmax=DEFAULT_GMAX,
+    smax=DEFAULT_SMAX,
+    constraints_order=None,
+    **constraints_kwargs,
 ):
-    """Display of 2D trajectory.
+    """Display 2D trajectories.
 
     Parameters
     ----------
     trajectory : array_like
         Trajectory to display.
-    size : float, optional
+    figsize : float, optional
         Size of the figure.
-    one_shot : bool, optional
-        If True, highlight the trajectory of the middle repetition.
-    constraints : bool, optional
-        If True, display the points where the gradients and slews are above the
-        default limits.
-    subfigure: plt.Axes or None, optional
-    resolution: Union[float, Tuple[float, ...]], optional
-        Resolution of MR image in m.
-        The default is DEFAULT_RESOLUTION.
+    one_shot : bool or int, optional
+        State if a specific shot should be highlighted in bold black.
+        If `True`, highlight the middle shot.
+        If `int`, highlight the shot at that index.
+        The default is `False`.
+    subfigure: plt.Figure or plt.SubFigure, optional
+        The figure where the trajectory should be displayed.
+        The default is `None`.
+    show_constraints : bool, optional
+        Display the points where the gradients and slew rates
+        are above the `gmax` and `smax` limits, respectively.
+        The default is `False`.
+    gmax: float, optional
+        The maximum constraint on the gradients in T/m.
+        The default is `DEFAULT_GMAX`.
+    smax: float, optional
+        The maximum constraint on the slew rates in T/m/ms.
+        The default is `DEFAULT_SMAX`.
+    constraint_order: int, str, optional
+        The norm order defining how the constraints are checked,
+        typically 2 or 1, following the `numpy.linalg.norm`
+        conventions on argument `ord`.
+        The default is None.
+    **kwargs
+        Acquisition parameters used to check on hardware constraints,
+        following the argument convention from
+        `mrinufft.trajectories.utils.compute_gradients_and_slew_rates`.
 
     Returns
     -------
     ax : plt.Axes
         Axes of the figure.
     """
+    displayConfig = DisplayConfig()
```

Comment:
no need for it, its a global of the module.
```suggestion
```

Reply:
Done